### ISSUE TITLE
CreateTryBlock cannot find finally method in yield return state machine

### ISSUE THREAD

**Issue by greenozon** [icsharpcode/ILSpy#2025]
ILSpy version 6.0.0.5752-preview4
save proj

```
 ---> ICSharpCode.ILSpy.AssertionFailedException: 
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message, String detailMessage) in offset 380
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message) in offset 9
   at System.Diagnostics.TraceInternal.Fail(String message) in offset 55
   at System.Diagnostics.Debug.Assert(Boolean condition) in offset 6
   at ICSharpCode.Decompiler.IL.ControlFlow.YieldReturnDecompiler.<>c__DisplayClass35_0.<ReconstructTryFinallyBlocks>g__CreateTryBlock|1(Block block, Int32 state) in offset 18
   at ICSharpCode.Decompiler.IL.ControlFlow.YieldReturnDecompiler.ReconstructTryFinallyBlocks(ILFunction iteratorFunction) in offset 270
   at ICSharpCode.Decompiler.IL.ControlFlow.YieldReturnDecompiler.Run(ILFunction function, ILTransformContext context) in offset 627
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 605
-- continuing with outer exception (ICSharpCode.Decompiler.DecompilerException) --
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 1125
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompile(IMethod method, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 253
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompile(ITypeDefinition typeDef, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 787
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompileTypes(IEnumerable`1 types, DecompileRun decompileRun, ITypeResolveContext decompilationContext, SyntaxTree syntaxTree) in offset 219
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileTypes(IEnumerable`1 types) in offset 217
   at ICSharpCode.Decompiler.CSharp.WholeProjectDecompiler.<>c__DisplayClass43_0.<WriteCodeFilesInProject>b__2(IGrouping`2 file) in offset 69
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1() in offset 589
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask) in offset 6
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object <p0>) in offset 134

```

https://www.sendspace.com/file/04zfhs

**Comment by siegfriedpammer** [comment]
The method token is `@06000050`

**Comment by siegfriedpammer** [comment]
Fixed by 4f4ca482be9c448768a8ce9c737d91ec7ec8a137 see #1950
